### PR TITLE
Feature/em 1283 assembly viewer tidyup

### PR DESCRIPTION
**Pull request by benouaer**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EM-1283
### Change description ###
Improving test coverage
**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: projects/dg-docassembly-webcomponent/src/lib/assembly/assembly-viewer/form-viewer/form-viewer.component.ts [hunk 1]
```diff
@@ -1,31 +1,60 @@
...
-      this.uiDefinition = this.assemblyService.getUIDefinition(changes.templateName.currentValue);
-    }
+  setOutputFormat(outputFormat: string) {
+    this.outputFormat = outputFormat;
+  }
+
+  onPreview() {
+    this.assemblyService
+      .generateDocument(this.outputFormat, this.templateName, this.templateData, this.documentUrl)
+      .subscribe(documentUrl => {
```
Comment (linusnorton) [inline projects/dg-docassembly-webcomponent/src/lib/assembly/assembly-viewer/form-viewer/form-viewer.component.ts:44]: This inline function is a bit large, please extract it to a private method

File: projects/dg-docassembly-webcomponent/src/lib/assembly/service/assembly.service.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+export const  ERROR = '--error--';
+
+@Injectable()
+export class AssemblyService {
+
+  constructor(private http: HttpClient) {}
+
+  getUIDefinition(templateName: string): Observable<FormlyFieldConfig[]> {
+    const encTemplateId = btoa(templateName);
+    return this.http.get<any>(`${uiDefinitionEndpoint}/${encTemplateId}`)
```
Comment (linusnorton) [inline projects/dg-docassembly-webcomponent/src/lib/assembly/service/assembly.service.ts:19]: one day I would like to stop doing <any> for the server responses, it shouldn't be that hard to create an interface for them. I understand that this just how it's done everywhere though

**Review by linusnorton — APPROVED** [review]

File: projects/dg-docassembly-webcomponent/src/lib/assembly/assembly-viewer/form-viewer/form-viewer.component.ts [hunk 3]
```diff
@@ -1,31 +1,60 @@
...
 @Component({
   selector: 'app-form-viewer',
   templateUrl: './form-viewer.component.html'
 })
-export class FormViewerComponent implements OnChanges {
+export class FormViewerComponent implements OnInit {
 
   uiDefinition: Observable<FormlyFieldConfig[]>;
   form = new FormGroup({});
+  outputFormat = 'PDF';
```
Comment (velimd) [inline projects/dg-docassembly-webcomponent/src/lib/assembly/assembly-viewer/form-viewer/form-viewer.component.ts:21]: is this the default value, as i believe, the app had the option to pass in multiple output formats.

File: projects/dg-docassembly-webcomponent/src/lib/assembly/assembly-viewer/form-viewer/form-viewer.component.ts [hunk 4]
```diff
@@ -1,31 +1,60 @@
...
+  documentUrl: string;
+  error: string;
+
   @Input() templateName: string;
   @Input() templateData: any;
+  @Input() outputFormats: string[];
```
Comment (velimd) [inline projects/dg-docassembly-webcomponent/src/lib/assembly/assembly-viewer/form-viewer/form-viewer.component.ts:27]: not sure if this is being used here.

**Pull request merged**